libs/withdraws/src/lib/withdraw-manager.spec.tsx

```diff
@@ -23,77 +47,84 @@ jest.mock('./use-get-withdraw-delay', () => ({
   },
 }));
 
-let props: WithdrawManagerProps;
+describe('WithdrawManager', () => {
+  let props: WithdrawManagerProps;
 
-beforeEach(() => {
-  props = {
-    assets: [generateAsset()],
-    accounts: [generateAccount()],
-    submit: jest.fn(),
-  };
-});
+  beforeEach(() => {
+    props = {
+      assets: [asset],
+      accounts: [generateAccount()],
+      submit: jest.fn(),
+      assetId: asset.id,
+    };
+  });
 
-const generateJsx = (props: WithdrawManagerProps) => (
-  <WithdrawManager {...props} />
-);
+  const generateJsx = (props: WithdrawManagerProps) => (
+    <WithdrawManager {...props} />
+  );
 
-it('calls submit if valid form submission', async () => {
-  render(generateJsx(props));
-  await act(async () => {
-    await submitValid();
-  });
-  expect(props.submit).toHaveBeenCalledWith({
-    amount: '1000',
-    asset: props.assets[0].id,
-    receiverAddress: ethereumAddress,
-    availableTimestamp: null,
+  it('calls submit if valid form submission', async () => {
+    render(generateJsx(props));
+    await act(async () => {
+      await submitValid();
+    });
+    await waitFor(async () => {
```

Comment:
chore (blocking): Some redundant code here can be tidied up, no need for waitFor, no need to await props.submit

```
render(generateJsx(props));
await act(async () => {
  await submitValid();
});
expect(props.submit).toHaveBeenCalledWith({
  amount: '1000',
  asset: props.assets[0].id,
  receiverAddress: ethereumAddress,
  availableTimestamp: null,
});
```

Reply:
updated